cranelift-filetests/src/function_runner.rs

```diff
@@ -76,13 +179,175 @@ impl FunctionRunner {
+    pub fn with_host_isa(self, flags: settings::Flags) -> Self {
+        let builder = host_isa_builder().expect("Unable to build a TargetIsa for the current host");
+        self.with_isa(builder.finish(flags))
+    }
+
+    /// Use the default host ISA.
+    pub fn with_default_host_isa(self) -> Self {
+        let flags = settings::Flags::new(settings::builder());
+        self.with_host_isa(flags)
+    }
+
+    /// Build a function runner from an action, a function and the ISA to run on.
+    pub fn finish(self) -> FunctionRunner {
+        FunctionRunner::new(self.function, self.isa.unwrap(), self.action)
+    }
+
+    /// Parse a function runner action from a comment string such as "; run print".
+    fn parse_action(comment: &str, ty: types::Type) -> FunctionRunnerAction {
+        let comment = comment.trim();
+        let comment = comment.trim_start_matches(";");
+        let comment = comment.trim_start();
+        let comment = comment.trim_start_matches("run");
+        let comment = comment.trim_start_matches(":");
+        let comment = comment.trim_start();
+
+        if comment.len() == 0 {
+            // Don't print test return value.
+            return FunctionRunnerAction::Test;
+        }
+
+        if comment == "print" {
+            // Print test return value, if return type is not b1, always fail the test.
+            return FunctionRunnerAction::Print;
+        }
+
+        // ";   run:  %fn() > 42 " is now "%fn() > 42".
+        let mut parts: Vec<&str> = comment.split_ascii_whitespace().collect();
+
+        if parts.len() < 3 {
+            panic!("Invalid syntax for test assertion, expected \"%fn() <operator> <value>\", found \"{}\"", comment);
+        }
+
+        // Last part must be a constant.
+        let expected = parse_constant_data(parts[parts.len() - 1], ty).unwrap();
+        parts.drain(parts.len() - 1..);
+
+        let operator = Self::parse_operator(parts[parts.len() - 1]).unwrap();
+        parts.drain(parts.len() - 1..);
+
+        if parts.len() != 1 {
+            // Arguments not supported.
+            panic!("Specifying arguments is not supported. \"%fn()\" <operator> <value>\", found \"{}\"", parts.join(" "));
+        }
+
+        let invocation = parts[0].trim();
+        let invocation = invocation.trim_start_matches("%fn(");
+        let invocation = invocation.trim_end_matches(")");
+        let invocation = invocation.trim();
+
+        if invocation.len() == 0 {
+            return FunctionRunnerAction::TestWithResult {
+                operator: operator,
+                expected: expected,
+            };
+        }
+
+        panic!(
+            "Expected \"%fn()\" <operator> <value>\", found \"{}\"",
+            parts.join(" ")
+        );
+    }
+
+    fn parse_operator(s: &str) -> Result<Operator, String> {
+        match s {
+            "==" => Ok(Operator::EQ),
+            "!=" => Ok(Operator::NEQ),
+            _ => Err(format!("Unsupported operator {}", s)),
```

Comment:
A minor re-factoring suggestion: elsewhere in cranelift I've seen this type of thing as `impl FromStr for Operator { ... }` so that we can just call `.parse()` up on line 320. Not a big deal but it keeps the operator implementation together.